### DISCUSSION TITLE
no visible DeviceName

### DISCUSSION THREAD

**Discussion by anglerCezary** [jziolkowski/tdm#205] [General]
Firstly - your @jziolkowski  project is great, I am his fan. After migrating from Linux Mint 20.3 to Mint 21, I lost my view DeviceName in TDM. the view of the other columns works fine.I have already tried removing and reinstalling the package but it doesn't fix the problem. I will be grateful if you guide me to the possible source of the problem.
P.S.
Przepraszam za mój angielski, korzystam z tłumacza internetowego.

**Comment by barbudor** [comment]
Could you please post a screenshot showing the problem as well as the content of your `tdm.cfg` file located in the TDM folder in your home folder (for example in Windows it is `C:\Users\<yourname>\TDM`, on Linux it is `/home/<yourname>/TDM`)

It is normally not possible to loose the "Device" column

**Reply by anglerCezary** [reply]
The column is visible. The names themselves are invisible.
File content` tdm.cfg`
```
[General]
autotelemetry=5000
connect_on_startup=true
hostname=192.168.2.192
password=
port=1883
username=ais
version=0.2.11
views_order="Home;Health;Firmware;Wifi;MQTT"
window_geometry=@ByteArray(\x1\xd9\xd0\xcb\0\x3\0\0\0\0\0\0\0\0\0\0\0\0\a\x7f\0\0\x4\xf\0\0\x2\x80\0\0\0\xb9\0\0\a\x7f\0\0\x3\x88\0\0\0\0\x2\0\0\0\a\x80\0\0\0\0\0\0\0%\0\0\a\x7f\0\0\x4\xf)

[Patterns]
0=%prefix%/%topic%/
1=dom/
2=%topic%/%prefix%/

[Views]
Firmware="Version;Core;SDK;ProgramSize;Free;OtaUrl"
Health="Uptime;BootCount;RestartReason;LoadAvg;Sleep;MqttCount;LinkCount;Downtime;RSSI"
Home="Module;Power;Color;LoadAvg;LinkCount;Uptime"
MQTT="Topic;FullTopic;CommandTopic;StatTopic;TeleTopic;FallbackTopic;GroupTopic"
Wifi="Hostname;Mac;IPAddress;Gateway;SSId;BSSId;Channel;RSSI;LinkCount;Downtime"
```
![obraz](https://user-images.githubusercontent.com/59963864/188328364-f499c259-b03b-463a-9286-0dc8951e96a3.png)

**Reply by barbudor** [reply]
Are you sure your device have a proper DeviceName ?
What are the version of your Tasmotas ?
Was it working before and did it suddenly stopped working ?
If yes, do you remember having changing something ? either in Tasmota or something else in your infrastructure ?

**Reply by anglerCezary** [reply]
I'm sure the problem arose after updating the Linux Mint release. All TDM functions work except for the name display itself. It has nothing to do with devices. MQTT is working properly.
![obraz](https://user-images.githubusercontent.com/59963864/188328972-81641e34-e227-42b9-94d4-6ccf280c392a.png)

**Reply by barbudor** [reply]
You can get other parameters from various versions of Tasmota so neither Tasmota config nor the TDM configuration seems to be the reason.

The upgrade of Mint may have upgraded a Python package used by TDM which is very sensistive to some version of PyQt5
```
PyQt5>=5.14.2,<6
```
Did you installedwith `pip install` ? 
If yes, can you type `pip list` and report the version of your `PyQt5` ?

**Reply by anglerCezary** [reply]
```
cezary@Lenovo ~> pip list | grep 'PyQt5'
PyQt5                  5.14.2
PyQt5-sip              12.11.0
cezary@Lenovo ~> 
```

**Reply by barbudor** [reply]
OK so that's not the reason
Running out of ideas for now

**Comment by anglerCezary** [comment]
I see a repeated error in the console:
```
Traceback (most recent call last):
  File "/home/cezary/.local/lib/python3.10/site-packages/Util/models.py", line 281, in paint
    p.drawPixmap(option.rect.x() + 2, option.rect.y() + px_y, px.scaled(24, 24))
TypeError: arguments did not match any overloaded call:
  drawPixmap(self, QRectF, QPixmap, QRectF): argument 1 has unexpected type 'int'
  drawPixmap(self, QRect, QPixmap, QRect): argument 1 has unexpected type 'int'
  drawPixmap(self, Union[QPointF, QPoint], QPixmap): argument 1 has unexpected type 'int'
  drawPixmap(self, QPoint, QPixmap): argument 1 has unexpected type 'int'
  drawPixmap(self, QRect, QPixmap): argument 1 has unexpected type 'int'
  drawPixmap(self, int, int, QPixmap): argument 2 has unexpected type 'float'
  drawPixmap(self, int, int, int, int, QPixmap): argument 2 has unexpected type 'float'
  drawPixmap(self, int, int, int, int, QPixmap, int, int, int, int): argument 2 has unexpected type 'float'
  drawPixmap(self, int, int, QPixmap, int, int, int, int): argument 2 has unexpected type 'float'
  drawPixmap(self, Union[QPointF, QPoint], QPixmap, QRectF): argument 1 has unexpected type 'int'
  drawPixmap(self, QPoint, QPixmap, QRect): argument 1 has unexpected type 'int'
```

**Reply by barbudor** [reply]
This part of your configuration looks strange:
```
[Patterns]
0=%prefix%/%topic%/
1=dom/
2=%topic%/%prefix%/
```
The 2nd one `dom/` looks illegal
The 1st and 3rd are conflicting

**Reply by anglerCezary** [reply]
Please don't look for problem in MQTT communication. There is nothing strange about it for me. The `devices.cfg` file contains the names of my devices read properly.

**Comment by moorep999** [comment]
I'm experiencing exactly the same problem after an upgrade to Mint 21.
* Installed with pip install
*  PyQt5 is same 5.14.2

Same error with the displaying device names. If I mouse over the entry it shows the name in the popup next to the cursor, but not in the devices field.

I have also checked that the devices.cfg file is ok - they have all the entries I'd expect with the correct device name. I renamed the devices.cfg file and restarted tdmgr which populated the devices list again, and they're blank as before.

As Cezar8421 has said, it all appears functional except displaying the device name.

**Comment by jziolkowski** [comment]
The underlying issue should be fixed in the current develop, give it a try

**Comment by anglerCezary** [comment]
Hello, I would like to be able to use TDM but there is still no solution to the problem of not displaying device names. Can any fix be expected? I am currently using Linux Mint 21.2.
```
pip list | grep 'PyQt5'
PyQt5                  5.14.2
PyQt5-Qt5              5.15.2
PyQt5-sip              12.11.0
```

**Comment by arendst** [comment]
Latest dev version (0.3) on Linux Mint 21.2 Cinnamon fixed it for me.
